File: BPPC12_2.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    public static void main(String[] args) {
        
        Scanner sc=new Scanner(System.in);
        long N=sc.nextLong();
        if(N>=2 && N<=Math.pow(10,12))
        {
            long x,y;
            long min=N+2;
            long minX=0;
            long minY=0;
            for(long i=1;i<=N/4;i++)
            {
                if(N%i==0)
                {
                    x=N/i;
                    y=i;
                    if((x+y)<min){
                        minX=x;
                        minY=y;
                        min=x+y;}
                }
            }
            System.out.println(minX+minY-2);
        }
        
    }
}

File: BPPC12_2.py
N=int(input())
x=0
y=0
minX=0
minY=0
min=N+2
i=1

while i<=int(N/4):
    if(N%i==0):
        x=int(N/i)
        y=i
        if(x+y)<min:
            min=x+y
            minX=x
            minY=y
    i=i+1

print(minX+minY-2)
    



